pkg/apis/core/validation/shoot_test.go

```diff
@@ -396,15 +394,66 @@ var _ = Describe("Shoot Validation Tests", func() {
 					"foo": "bar",
 				}
 
-				errorList := ValidateShootUpdate(newShoot, shoot)
+				errorList := ValidateShootHAControlPlaneUpdate(newShoot, shoot)
 
 				Expect(errorList).To(ConsistOf(
 					PointTo(MatchFields(IgnoreExtras, Fields{
 						"Type":  Equal(field.ErrorTypeInvalid),
-						"Field": Equal("metadata.annotations[alpha.control-plane.shoot.gardener.cloud/high-availability]"),
+						"Field": Equal(fmt.Sprintf("metadata.annotations[%s]", v1beta1constants.ShootAlphaControlPlaneHighAvailability)),
+					})),
+				))
+			})
+
+			It("should pass as Shoot ControlPlane Spec has not changed", func() {
+				shoot.Spec.ControlPlane = &core.ControlPlane{HighAvailability: core.HighAvailability{FailureTolerance: core.FailureTolerance{FailureToleranceType: core.FailureToleranceTypeZone}}}
+				newShoot := prepareShootForUpdate(shoot)
+				errorList := ValidateShootHAControlPlaneUpdate(newShoot, shoot)
+				Expect(errorList).To(HaveLen(0))
+			})
+
+			It("should forbid to change the Shoot ControlPlane spec", func() {
+				shoot.Spec.ControlPlane = &core.ControlPlane{HighAvailability: core.HighAvailability{FailureTolerance: core.FailureTolerance{FailureToleranceType: core.FailureToleranceTypeZone}}}
+				newShoot := prepareShootForUpdate(shoot)
+				newShoot.Spec.ControlPlane = &core.ControlPlane{HighAvailability: core.HighAvailability{FailureTolerance: core.FailureTolerance{FailureToleranceType: core.FailureToleranceTypeNode}}}
+
+				errorList := ValidateShootHAControlPlaneUpdate(newShoot, shoot)
+				Expect(errorList).To(ConsistOf(
+					PointTo(MatchFields(IgnoreExtras, Fields{
+						"Type":     Equal(field.ErrorTypeInvalid),
+						"BadValue": Equal(core.FailureToleranceTypeNode),
+						"Field":    Equal("spec.ControlPlane.HighAvailability.FailureTolerance.FailureToleranceType"),
 					})),
 				))
 			})
+
+			It("should forbid to unset of Shoot ControlPlane", func() {
+				shoot.Spec.ControlPlane = &core.ControlPlane{HighAvailability: core.HighAvailability{FailureTolerance: core.FailureTolerance{FailureToleranceType: core.FailureToleranceTypeZone}}}
+				newShoot := prepareShootForUpdate(shoot)
+				newShoot.Spec.ControlPlane = nil
+
+				errorList := ValidateShootHAControlPlaneUpdate(newShoot, shoot)
+
+				Expect(errorList).To(ConsistOf(
+					PointTo(MatchFields(IgnoreExtras, Fields{
+						"Type":  Equal(field.ErrorTypeForbidden),
+						"Field": Equal("spec.ControlPlane"),
+					})),
+				))
+			})
+			It("cannot set both HAControlPlane annotation and Shoot ControlPlane Spec", func() {
```

Comment:
```suggestion

			It("cannot set both HAControlPlane annotation and Shoot ControlPlane Spec", func() {
```

Leave an empty line between `It`s, `Describe`s, etc. for readability